src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/DeleteCommand.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Logging;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal static class DeleteCommand
+    {
+        public static void Register(CommandLineApplication app, Func<ILogger> getLogger)
+        {
+            app.Command("delete", delete =>
+            {
+                delete.Description = Strings.Delete_Description;
+
+                var source = delete.Option(
+                    "-s|--source <source>",
+                    Strings.Source_Description,
+                    CommandOptionType.SingleValue);
+
+                var nonInteractive = delete.Option(
+                    "--non-interactive",
+                    Strings.NonInteractive_Description,
+                    CommandOptionType.NoValue);
+
+                var apikey = delete.Option(
+                    "-k|--api-key <apiKey>",
+                    Strings.ApiKey_Description,
+                    CommandOptionType.SingleValue);
+
+                var arguments = delete.Argument(
+                    "[root]",
+                    Strings.Delete_PackageIdAndVersion_Description,
+                    multipleValues: true);
+
+                delete.OnExecute(async () =>
+                {
+                    if (arguments.Values.Count < 2)
+                    {
+                        throw new ArgumentException(Strings.Delete_MissingArguments);
+                    }
+
+                    string packageId = arguments.Values[0];
+                    string packageVersion = arguments.Values[1];
+                    string sourcePath = source.Value();
+                    string apiKeyValue = apikey.Value();
+                    bool nonInteractiveValue = nonInteractive.HasValue();
+
+                    ISettings settings = Settings.LoadDefaultSettings(Directory.GetCurrentDirectory(), configFileName: null, machineWideSettings: null);
```

Comment:
This may need machine wide settings incase the default source or api key was defined there. @zhili1208 has fixed several similar issue in xplat with this and would know the best.